### PR TITLE
[issue_tracker] Fixes some of the functionality of the module

### PR DESCRIPTION
**Pull request by maltheism**
### Brief summary of changes

Fixes a lot of issue_tracker bugs and the module isn't functional in current state. I'm just publishing this PR right now because it solves a lot of the bugs for geting the module close to working.

Some of the issues: Checkbox not working, Site not being populated in the table, clicking an issue not populating site and I can't remember everything because it was frustrating to debug & fix.

### This resolves issue...

- [ ] Redmine? #

- [x] Github? #4582

### To test this change...

- [x] todo

### CODE REVIEW

**Comment by maltheism** [comment]
Please don't review just yet.

**Comment by maltheism** [comment]
Hey @zaliqarosli, there isn't any console.logs and I thought we now commit the package-lock.json file. I'm looking at the differences between mine vs the old and it seems that https is now used in mine which is actually better.

File: modules/issue_tracker/jsx/IssueForm.js [hunk 1]
```diff
@@ -273,6 +274,13 @@ class IssueForm extends Component {
     $.ajax(this.props.DataURL, {
       dataType: 'json',
       success: function(data) {
+        console.log(data);
```
Comment (zaliqarosli) [inline modules/issue_tracker/jsx/IssueForm.js]: console.log!

**Comment by zaliqarosli** [comment]
@maltheism yeahh unsure about the package-lock. i forgot what we decided after that update PR. @johnsaigle do you remember?

File: modules/issue_tracker/jsx/IssueForm.js (hunk 1 above)
Comment (maltheism) [reply modules/issue_tracker/jsx/IssueForm.js]: @zaliqarosli hah missed it thanks :)

**Comment by maltheism** [comment]
@zaliqarosli https://github.com/aces/Loris/issues/4408

**Comment by johnsaigle** [comment]
My preference would be to submit updates to `.lock` files in their own PRs. If you attach them to other PRs it can be a little harder to go back and track changes and discussions.

I don't really deal much with the JS side though so I think you should just go with whatever works best for you all.

**Comment by maltheism** [comment]
Hey @zaliqarosli & @johnsaigle my opinion is we just commit the package-lock.json and literally ignore it in PRs. Reason is every time I run `make` or `npm run compile` the file is made and when I go to add files I'm just doing `git add *` if I'm having a happy day with my workflow. Having to dodge the file by adding all files modified manually is a joke of a life. Thus, right before publishing release we checkout 21.0-release, make a pr that just runs `npm run compile` and commit the package-lock.json file. Otherwise it would be nice to just ignore the file. Whatever the solution happens to be.. I want my git add * command or I'll be more sad in life.

**Comment by zaliqarosli** [comment]
still one more error left on the 'Create New Issue' page

Screenshot 2019-05-30 10 22 19

**Comment by maltheism** [comment]
Hey @zaliqarosli, I'm not getting that error when making an issue with this PR.

**Comment by zaliqarosli** [comment]
module seems to be working fine now, need to do a proper testing. whatever the solution is for package-lock.json, i don't think it should be included in this PR so i'm still requesting for it to be removed from this PR. if you want to use `git add *` functionality, you can also easily do `git checkout -- package-lock.json` after. i personally don't use git add * so this is not any issue i'm invested in, sorry!

**Comment by maltheism** [comment]
Hey @zaliqarosli I'm leaving the package-lock.json in this PR.

**Comment by driusan** [comment]
Please split this into multiple PRs.. it can't really be reviewed right now, because it doesn't describe what it's fixing, and it's going to conflict with other PRs that are fixing individual problems with the issue tracker.

**Comment by zaliqarosli** [comment]
you can remove the package-lock.json file from this PR now because https://github.com/aces/Loris/pull/4791 takes care of that

**Comment by driusan** [comment]
As far as I can tell, everything that this PR was trying to do has been fixed in other more targeted PRs now.

**Pull request closed without merging**